### PR TITLE
Add new chat listener

### PR DESCRIPTION
**Pull request by builder-247**
* TODO -  test with odota/core

See https://github.com/odota/web/issues/2747

### CODE REVIEW

File: src/main/java/opendota/Parse.java [hunk 1]
```diff
@@ -306,7 +307,18 @@ public void onChatEvent(Context ctx, CDOTAUserMsg_ChatEvent message) {
         entry.value = value;
         output(entry);
     }
-    
+
+    // New chat event
+    @OnMessage(CDOTAUserMsg_ChatMessage.class)
+    public void onAllChatMessage(Context ctx, CDOTAUserMsg_ChatMessage message) {
+        String type = String.valueOf(message.getChannelType());
```
Comment (howardchung) [inline src/main/java/opendota/Parse.java]: What are the possible types here? If we want to be backward compatible I think we have to map 11 to "chat"

**Comment by builder-247** [comment]
Looks like `web` uses the `unit` property, any ideas how to convert slot to playername?

File: src/main/java/opendota/Parse.java [hunk 2]
```diff
@@ -306,7 +307,18 @@ public void onChatEvent(Context ctx, CDOTAUserMsg_ChatEvent message) {
...
         output(entry);
     }
-    
+
+    // New chat event
+    @OnMessage(CDOTAUserMsg_ChatMessage.class)
+    public void onAllChatMessage(Context ctx, CDOTAUserMsg_ChatMessage message) {
+        int type = message.getChannelType();
+        Entry entry = new Entry(time);
+        entry.slot = message.getSourcePlayerId();
```
Comment (howardchung) [inline src/main/java/opendota/Parse.java:316]: can you verify this is 0-4 and 128-132?

File: src/main/java/opendota/Parse.java [hunk 3]
```diff
@@ -306,7 +307,18 @@ public void onChatEvent(Context ctx, CDOTAUserMsg_ChatEvent message) {
...
+        entry.type = (type == 11) ? "chat" : String.valueOf(type);
```
Comment (howardchung) [inline src/main/java/opendota/Parse.java:317]: did you see any non-11 events get emitted in a test replay?

**Comment by howardchung** [comment]
I don't think we have to convert, we could just assign to a player based on slot and use the name we already have rather than in the chat event

File: src/main/java/opendota/Parse.java (hunk 3 above)
Comment (builder-247) [reply src/main/java/opendota/Parse.java:317]: nope - my guess is the other types (teamchat/coach chat) are stripped from replays

File: src/main/java/opendota/Parse.java (hunk 2 above)
Comment (builder-247) [reply src/main/java/opendota/Parse.java:316]: 0-9 like the old chat

**Comment by builder-247** [comment]
> I don't think we have to convert, we could just assign to a player based on slot and use the name we already have rather than in the chat event

yeah we should prob just make another hashmap, `slot_to_name`

**Comment by howardchung** [comment]
I don't think we even need to do that. The existing code just uses unit as a backup, we display the user's in-game name if we have one:
https://github.com/odota/web/blob/master/src/components/Match/Chat/Chat.jsx#L482

File: src/main/java/opendota/Parse.java (hunk 2 above)
Comment (howardchung) [reply src/main/java/opendota/Parse.java:316]: that is probably fine as long as it matches the old `        entry.slot = getEntityProperty(e, "m_iPlayerID", null);`

**Comment by howardchung** [comment]
Did you want to verify with odota/core before I merge this? Looks good to me

**Comment by builder-247** [comment]
> Did you want to verify with odota/core before I merge this? Looks good to me

Yeah it produces chat events now so it should be fine

**Pull request merged**